altair/vegalite/v5/schema/_config.py

```diff
@@ -0,0 +1,1055 @@
+    tooltip: str | bool | None | float | TooltipContent
+    url: str
+    width: float
+    x: str | float
+    x2: str | float
+    y: str | float
+    y2: str | float
+
+
+class TitleConfigKwds(TypedDict, total=False):
+    """Placeholder doc."""
+
+    align: Align_T
+    anchor: TitleAnchor_T
+    angle: float
+    aria: bool
+    baseline: str | Baseline_T
+    color: str | None | ColorName_T
+    dx: float
+    dy: float
+    font: str
+    fontSize: float
+    fontStyle: str | FontStyle
+    fontWeight: FontWeight_T
+    frame: str | TitleFrame_T
+    limit: float
+    lineHeight: float
+    offset: float
+    orient: TitleOrient_T
+    subtitleColor: str | None | ColorName_T
+    subtitleFont: str
+    subtitleFontSize: float
+    subtitleFontStyle: str | FontStyle
+    subtitleFontWeight: FontWeight_T
+    subtitleLineHeight: float
+    subtitlePadding: float
+    zindex: float
+
+
+class FormatConfigKwds(TypedDict, total=False):
+    """Placeholder doc."""
+
+    normalizedNumberFormat: str
+    normalizedNumberFormatType: str
+    numberFormat: str
+    numberFormatType: str
+    timeFormat: str
+    timeFormatType: str
+
+
+class ViewConfigKwds(TypedDict, total=False):
+    """Placeholder doc."""
+
+    clip: bool
+    continuousHeight: float
+    continuousWidth: float
+    cornerRadius: float
+    cursor: Cursor_T
+    discreteHeight: float
+    discreteWidth: float
+    fill: str | None | ColorName_T
+    fillOpacity: float
+    opacity: float
+    step: float
+    stroke: str | None | ColorName_T
+    strokeCap: StrokeCap_T
+    strokeDash: Sequence[float]
+    strokeDashOffset: float
+    strokeJoin: StrokeJoin_T
+    strokeMiterLimit: float
+    strokeOpacity: float
+    strokeWidth: float
+
+
+# TODO: Non-`TypedDict` arg
+
+
+class ScaleInvalidDataConfigKwds(TypedDict, total=False):
+    """Placeholder doc."""
+
+    angle: str | ScaleInvalidDataShowAsValueangle
+    color: str | ScaleInvalidDataShowAsValuecolor
+    fill: str | ScaleInvalidDataShowAsValuefill
+    fillOpacity: str | ScaleInvalidDataShowAsValuefillOpacity
+    opacity: str | ScaleInvalidDataShowAsValueopacity
+    radius: str | ScaleInvalidDataShowAsValueradius
+    shape: str | ScaleInvalidDataShowAsValueshape
+    size: str | ScaleInvalidDataShowAsValuesize
+    stroke: str | ScaleInvalidDataShowAsValuestroke
+    strokeDash: str | ScaleInvalidDataShowAsValuestrokeDash
+    strokeOpacity: str | ScaleInvalidDataShowAsValuestrokeOpacity
+    strokeWidth: str | ScaleInvalidDataShowAsValuestrokeWidth
+    theta: str | ScaleInvalidDataShowAsValuetheta
+    x: str | ScaleInvalidDataShowAsValuex
+    xOffset: str | ScaleInvalidDataShowAsValuexOffset
+    y: str | ScaleInvalidDataShowAsValuey
+    yOffset: str | ScaleInvalidDataShowAsValueyOffset
+
+
+class ThemeConfig(TypedDict, total=False):
+    """Placeholder doc."""
+
+    arc: RectConfigKwds
+    area: AreaConfigKwds
+    aria: Any  # TODO
+    autosize: Any  # TODO
+    axis: AxisConfigKwds
```

Comment:
Top-level `TypedDict`.

Anything labelled like `    aria: Any  # TODO` means that it is a simple annotation.

Leaving those until all other issues are resolved